coverage.c

```diff
@@ -0,0 +1,590 @@
+                    "  -q <int>          base quality threshold [0]\n"
+                    "  -Q <int>          mapping quality threshold [0]\n"
+                    "  -H                print a file header\n"
+                    "  -m                show histogram instead of tabular output\n"
+                    "  -w <int>          number of bins in histogram. Set to 0 to use terminal width [50]\n"
+                    "  -r <chr:from-to>  Show region on chromosome chr. \n"
+                    "  -h                help (this page)\n"
+                    "  -v                version of this command\n"
+                    "\nGlobal options:\n");
+
+    sam_global_opt_help(stderr, "-.--.-");
+
+    fprintf(stderr, 
+            "\nThe tabular output is a simple tab-seperate table with 7 columns:\n"
+                    "  1. Reference name / chromosome\n"
+                    "  2. Start position\n"
+                    "  3. End position (or sequence length)\n"
+                    "  4. Number reads aligned to the region\n"
+                    "  5. Number of covered bases with depth >= 1\n"
+                    "  6. Percent of covered bases\n"
+                    "  7. Average coverage depth\n"
+                    "  8. Average baseQ in covered region\n"
+                    "  9. Average mapQ of selected reads\n"
+                    );
+
+    return status;
+}
+
+static int version(int status) {
+    fprintf(stderr,
+    "┌┐ ┌─┐┌┬┐┌─┐┌─┐┬  ┬\n"
+    "├┴┐├─┤││││  │ │└┐┌┘\n"
+    "└─┘┴ ┴┴ ┴└─┘└─┘ └┘   v%s\n", VERSION);
+    return status;
+}
+
+char* centerText(char *text, char *buf, int width) {
+    int len = strlen(text);
+    int padding = (width - len) / 2;
+    int padding_ex = (width - len) % 2;
+    if (padding >= 1) 
+        sprintf(buf, " %*s%*s", len+padding, text, padding-1+padding_ex, " ");
+    return(buf);
+}
+
+char* readable_bps(double base_pairs, char *buf) {
+    const char* units[] = {"", "K", "M", "G", "T"};
+    int i = 0;
+    while (base_pairs >= 1000) {
+        base_pairs /= 1000;
+        i++;
+    }
+    sprintf(buf, "%.*f%s", i, base_pairs, units[i]);
+    return buf;
+}
+
+void set_read_counts(bam_aux_t **data, stats_aux_t *stats, int n_bam_files) {
+    int i;
+    stats->n_reads = 0;
+    stats->n_selected_reads = 0;
+    stats->summed_mapQ = 0;
+    for (i = 0; i < n_bam_files && data[i]; ++i) {
+        stats->n_reads += data[i]->n_reads;
+        stats->n_selected_reads += data[i]->n_selected_reads;
+        stats->summed_mapQ += data[i]->summed_mapQ;
+        data[i]->n_reads = 0;
+        data[i]->n_selected_reads = 0;
+        data[i]->summed_mapQ = 0;
+    }
+}
+
+// UTF8 specifies block characters in eights going from \u2581 (lower one eight block) to \u2588 (full block)
+//   https://en.wikipedia.org/wiki/Block_Elements
+const char* BLOCK_CHARS[8] = {"\u2581", "\u2582", "\u2583", "\u2584", "\u2585", "\u2586", "\u2587", "\u2588"};
```

Comment:
This code should work portably:
```c++
	/** Unicode bar characters. */
	static const char* BLOCK_CHARS[10] = {
		" ", "_",
		"\342\226\201", // 9601
		"\342\226\202", // 9602
		"\342\226\203", // 9603
		"\342\226\204", // 9604
		"\342\226\205", // 9605
		"\342\226\206", // 9606
		"\342\226\207", // 9607
		"\342\226\210", // 9608
	};
```
See https://github.com/bcgsc/abyss/blob/d4b4b5d3091d90a4967180d987bd7168dbf04585/Common/Histogram.cpp#L71-L82

Reply:
Fixed w/ @jmarshall 's `\xFF` version